auslib/blobs/gmp.py

```diff
@@ -61,6 +61,15 @@ def getInnerXML(self, updateQuery, update_type, whitelistedDomains, specialForce
 
         return vendorXML
 
+    def validate(self, *args, **kwargs):
+        Blob.validate(self, *args, **kwargs)
+        hashes = dict(SHA1=40, SHA256=64, SHA512=128)
```

Comment:
I think the function for validating whether a hash is the correct length or not doesn't need to live in Blob - there's nothing inherently Blob specific about it. I would suggest putting it somewhere in https://github.com/mozilla/balrog/tree/master/auslib/util.